### PR TITLE
Implement produce backup collection

### PR DESCRIPTION
**Pull request by neha-Gupta1**
This will go in once -  https://github.com/alcionai/corso/pull/4082 and https://github.com/alcionai/corso/pull/4050/ gets merged. Its a draft PR. Just for tracking as of now. 

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #<issue>

#### Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/groups/handler.go
Comment (ryanfkeepers) [inline src/internal/m365/collection/groups/handler.go:1]: Please use the `handlers.go` file.

File: src/pkg/backup/details/iteminfo.go [hunk 1]
```diff
@@ -37,6 +37,9 @@ const (
 
 	// Folder Management(30x)
 	FolderItem ItemType = 306
+
+	// GroupChannel(40x)
+	GroupChannel ItemType = 407
```
Comment (ryanfkeepers) [inline src/pkg/backup/details/iteminfo.go:42]:
```suggestion
	GroupChannel ItemType = 400
```

**Review by ryanfkeepers — COMMENTED** [review]
pausing this review midway through.  There's too many changes that are conflicting, or potentially conflict, with other PRs out there.  The only way to resolve that, really, is to make sure this is based on the proper parents.  IE: don't copy code in from other PRs.  This branch should be based on your API PR branch, and that branch should be based on either the current state of main, or some other feature branch like https://github.com/alcionai/corso/pull/4093.  But as it currently stands, there's too much chaff in the code to properly review, and I'm unable to separate the changes from the copied code.

**Comment by neha-Gupta1** [comment]
> is branch should be based on your API PR branch, and that br

I agree with you @ryanfkeepers. Let https://github.com/alcionai/corso/pull/4068 go in. And I will re-base my changes according to this PR then. Or I can go any better way which you suggest.

**Comment by ryanfkeepers** [comment]
closing; changes were ported to other PRs

**Pull request closed without merging**